src/runtime/node/console/index.ts

```diff
@@ -0,0 +1,230 @@
+) => {
+  if (globalThis.console && globalThis.console.log) {
+    return globalThis.console.log(...args);
+  }
+  createNotImplementedError("console.log");
+};
+
+export const profile: typeof console.profile = (
+  ...args: Parameters<(typeof globalThis)["console"]["profile"]>
+) => {
+  if (globalThis.console && globalThis.console.profile) {
+    return globalThis.console.profile(...args);
+  }
+  createNotImplementedError("console.profile");
+};
+
+export const profileEnd: typeof console.profileEnd = (
+  ...args: Parameters<(typeof globalThis)["console"]["profileEnd"]>
+) => {
+  if (globalThis.console && globalThis.console.profileEnd) {
+    return globalThis.console.profileEnd(...args);
+  }
+  createNotImplementedError("console.profileEnd");
+};
+
+export const table: typeof console.table = (
+  ...args: Parameters<(typeof globalThis)["console"]["table"]>
+) => {
+  if (globalThis.console && globalThis.console.table) {
+    return globalThis.console.table(...args);
+  }
+  createNotImplementedError("console.table");
+};
+
+export const time: typeof console.time = (
+  ...args: Parameters<(typeof globalThis)["console"]["time"]>
+) => {
+  if (globalThis.console && globalThis.console.time) {
+    return globalThis.console.time(...args);
+  }
+  createNotImplementedError("console.time");
+};
+
+export const timeEnd: typeof console.timeEnd = (
+  ...args: Parameters<(typeof globalThis)["console"]["timeEnd"]>
+) => {
+  if (globalThis.console && globalThis.console.timeEnd) {
+    return globalThis.console.timeEnd(...args);
+  }
+  createNotImplementedError("console.timeEnd");
+};
+
+export const timeLog: typeof console.timeLog = (
+  ...args: Parameters<(typeof globalThis)["console"]["timeLog"]>
+) => {
+  if (globalThis.console && globalThis.console.timeLog) {
+    return globalThis.console.timeLog(...args);
+  }
+  createNotImplementedError("console.timeLog");
+};
+
+export const timeStamp: typeof console.timeStamp = (
+  ...args: Parameters<(typeof globalThis)["console"]["timeStamp"]>
+) => {
+  if (globalThis.console && globalThis.console.timeStamp) {
+    return globalThis.console.timeStamp(...args);
+  }
+  createNotImplementedError("console.timeStamp");
+};
+
+export const trace: typeof console.trace = (
+  ...args: Parameters<(typeof globalThis)["console"]["trace"]>
+) => {
+  if (globalThis.console && globalThis.console.trace) {
+    return globalThis.console.trace(...args);
+  }
+  createNotImplementedError("console.trace");
+};
+
+export const warn: typeof console.warn = (
+  ...args: Parameters<(typeof globalThis)["console"]["warn"]>
+) => {
+  if (globalThis.console && globalThis.console.warn) {
+    return globalThis.console.warn(...args);
+  }
+  createNotImplementedError("console.warn");
+};
+
+export const Console: typeof console.Console =
+  globalThis.console && globalThis.console.Console
+    ? globalThis.console.Console
+    : mock.__createMock__("console.Console");
```

Comment:
You can use nullish coalescing operator here to simplify the code:

```suggestion
export const Console: typeof console.Console =
    // Node.js exposes the Console constructor on the console instance 🤷🏼‍♂️
    // https://nodejs.org/api/console.html#class-console
    globalThis.console?.Console ?? mock.__createMock__("console.Console");
```

I'm however not convinced that we should try to grab the constructor from globalThis.console at all as that's a node.js-only thing and is not spec compliant, so this would only work in node.js environment and `nodeless` which this polyfill is trying to backfill. So it's most likely never going to be used, which makes me think we should just always return a mock. But I wonder what does @pi0 think about this one.

Reply:
That's true indeed it is not specific but also like this because allows to optionally polyfill by summing in global `console` as an escape hatch.

Reply:
Sounds good, I've changed it back to the original implementation that tries to use `global.console.Console` and falls back to a mock. LMK if this works for y'all